manifests/params.pp

```diff
@@ -21,11 +23,12 @@
 # @param candlepin_oauth_secret
 #   The oauth secret for Candlepin
 class katello::params (
-  Stdlib::Httpsurl $candlepin_url = "https://${facts['fqdn']}:8443/candlepin",
   Stdlib::Httpsurl $pulp_url = "https://${facts['fqdn']}/pulp/api/v2/",
   Stdlib::Httpsurl $crane_url = "https://${facts['fqdn']}:5000",
   Stdlib::Host $qpid_hostname = 'localhost',
   String[1] $candlepin_oauth_key = $katello::globals::candlepin_oauth_key,
   String[1] $candlepin_oauth_secret = $katello::globals::candlepin_oauth_secret,
+  Stdlib::Host $candlepin_host = 'localhost',
+  Stdlib::Httpsurl $candlepin_url = "https://${candlepin_host}:8443/candlepin"
```

Comment:
I don't know if this works. I've seen people use it, but never trusted it. Given this is the only way to deploy candlepin, I'd move the `$candlepin_url` part to the body of this class.

Reply:
This worked in my testing running these changes live with the installer. If its defined in the body, will other modules be able to access it?

Reply:
Yes, that still works.